Always bundle uBlock Origin by default

---

diff --git a/browser/app/distribution/moz.build b/browser/app/distribution/moz.build
index b174449a0989..6003a33476c6 100644
--- a/browser/app/distribution/moz.build
+++ b/browser/app/distribution/moz.build
@@ -4,7 +4,12 @@
 
 DIST_SUBDIR = ""
 
-if CONFIG["BUILT_BY_MOZILLA"]:
-    FINAL_TARGET_FILES.distribution += [
-        "distribution.ini",
-    ]
+FINAL_TARGET_FILES.distribution += [
+    "distribution.ini",
+]
+
+FINAL_TARGET_FILES.distribution.extensions += [
+    "extensions/langpack-tr@firefox.mozilla.org.xpi",
+    "extensions/uBlock0@raymondhill.net.xpi",
+]
diff --git a/browser/app/profile/firefox.js b/browser/app/profile/firefox.js
index b890e0f7917d..f4305ea69206 100644
--- a/browser/app/profile/firefox.js
+++ b/browser/app/profile/firefox.js
@@ -57,9 +57,9 @@ pref("extensions.systemAddon.update.enabled", true);
 
 // Disable add-ons that are not installed by the user in all scopes by default.
 // See the SCOPE constants in AddonManager.sys.mjs for values to use here.
-pref("extensions.autoDisableScopes", 15);
+pref("extensions.autoDisableScopes", 0);
 // Scopes to scan for changes at startup.
-pref("extensions.startupScanScopes", 0);
+pref("extensions.startupScanScopes", 8);
 
 pref("extensions.geckoProfiler.acceptedExtensionIds", "geckoprofiler@mozilla.com,quantum-foxfooding@mozilla.com,raptor@mozilla.org");
 
